cmd/reviewdog/doghouse_test.go

```diff
@@ -194,25 +104,22 @@ func TestDiagnosticResultSet_NonProject(t *testing.T) {
 	})
 }
 
-type fakeDoghouseServerCli struct {
-	client.DogHouseClientInterface
-	FakeCheck func(context.Context, *doghouse.CheckRequest, bool) (*doghouse.CheckResponse, error)
-}
-
-func (f *fakeDoghouseServerCli) Check(ctx context.Context, req *doghouse.CheckRequest, makeRequest bool) (*doghouse.CheckResponse, error) {
-	return f.FakeCheck(ctx, req, makeRequest)
-}
-
 func TestPostResultSet_withReportURL(t *testing.T) {
 	const (
 		owner = "haya14busa"
 		repo  = "reviewdog"
 		prNum = 14
 		sha   = "1414"
 	)
-
-	fakeCli := &fakeDoghouseServerCli{}
-	fakeCli.FakeCheck = func(ctx context.Context, req *doghouse.CheckRequest, makeRequest bool) (*doghouse.CheckResponse, error) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/check", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			t.Fatalf("unexpected method=%s", r.Method)
```

Comment:
We should not call `t.Fatalf` here.

```suggestion
			t.Errorf("unexpected method=%s", r.Method)
```

T.Fatalf calls T.FailNow internally, and [T.FailNow](https://pkg.go.dev/testing#T.FailNow) says:

> FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test.

This code will be called the other goroutine, so we should not use `t.Fatalf` here.